ledger-tool/src/main.rs

```diff
@@ -2980,16 +2980,32 @@ fn main() {
                     value_t_or_exit!(arg_matches, "snapshot_slot", Slot)
                 };
 
-                assert!(
-                    blockstore.meta(snapshot_slot).unwrap().is_some(),
-                    "snapshot slot doesn't exist"
-                );
+                if blockstore.meta(snapshot_slot).unwrap().is_none() {
+                    // Find the highest full slot that is less than or equal to the snapshot slot, if it exists
+                    let most_recent_slot_before_snapshot_slot = blockstore
+                        .slot_meta_iterator(0)
+                        .ok()
+                        .map(|r| {
+                            r.filter_map(|meta| {
+                                (meta.0 <= snapshot_slot && meta.1.is_full()).then_some(meta.0)
+                            })
+                            .last()
+                        })
+                        .unwrap_or_default();
```

Comment:
So the slot found with this method could differ from the last actual slot that we could make a snapshot at. Recall that in the `slot_range_connected()`, we traverse the `next_slots` field to ensure we stay on the main fork.

Remind me, this check was added to the `create-snapshot` function specifically because we needed to relax the `halt-at-slot`-must-exist requirement for commands like `verify`, right?

Reply:
> So the slot found with this method could differ from the last actual slot that we could make a snapshot at. Recall that in the slot_range_connected(), we traverse the next_slots field to ensure we stay on the main fork.

I recall that, but we'll won't find anything that the requested snapshot slot is connected to, since it doesn't exist, right?

>Remind me, this check was added to the create-snapshot function specifically because we needed to relax the halt-at-slot-must-exist requirement for commands like verify, right?
Yeah, we don't need the exact slot for verify and other commands, but do for `create-snapshot`

Reply:
> I recall that, but we'll won't find anything that the requested snapshot slot is connected to, since it doesn't exist, right?

Imagine we have the following slots:
```
4 (existing snapshot slot)
5 (parent = 4)
6 (parent = 3)
7 (slot doesn't exists)
```
Suppose someone requests a snapshot at slot 7.
- With the code in this PR, the emitted error will say the largest slot is 6.
- If the user then tries to create a snapshot at slot 6, they'll run into the other error that tells them there is no path to 6.

Reply:
That's valid. I'll have to re-work this, we don't have the `starting_slot` here.

I can either: 
* Factor out the code for getting the starting_slot, and iterate via connected slots to the max slot
* or remove the nearest slot part of this message

Reply:
removed it